doc/metadata_routing.rst

```diff
@@ -61,88 +83,76 @@ in this section require the following imports and data::
 
 Note that in this example, ``my_weights`` is passed to both the scorer and
 :class:`~linear_model.LogisticRegressionCV`.
 
-Error handling: if ``props={"sample_weigh": my_weights, ...}`` were passed
+Error handling: if ``params={"sample_weigh": my_weights, ...}`` were passed
 (note the typo), :func:`~model_selection.cross_validate` would raise an error,
 since ``sample_weigh`` was not requested by any of its underlying objects.
 
 Weighted scoring and unweighted fitting
 ---------------------------------------
 
 When passing metadata such as ``sample_weight`` around, all ``sample_weight``
-:term:`consumers <consumer>` require weights to be either explicitly requested
-or not requested (i.e. ``True`` or ``False``) when used in another
-:term:`router` such as a :class:`~pipeline.Pipeline` or a ``*GridSearchCV``. To
-perform an unweighted fit, we need to configure
-:class:`~linear_model.LogisticRegressionCV` to not request sample weights, so
+:term:`consumers <consumer>` require weights to be either explicitly requested or not
+requested (i.e. ``True`` or ``False``). Thus, to perform an unweighted fit, we need to
+configure :class:`~linear_model.LogisticRegressionCV` to not request sample weights, so
 that :func:`~model_selection.cross_validate` does not pass the weights along::
 
-  >>> weighted_acc = make_scorer(accuracy_score).set_score_request(
-  ...     sample_weight=True
-  ... )
-  >>> lr = LogisticRegressionCV(
-  ...     cv=GroupKFold(), scoring=weighted_acc,
-  ... ).set_fit_request(sample_weight=False)
+  >>> weighted_acc = make_scorer(accuracy_score).set_score_request(sample_weight=True)
+  >>> lr = LogisticRegressionCV(cv=GroupKFold(), scoring=weighted_acc,)
+  >>> lr.set_fit_request(sample_weight=False)
   >>> cv_results = cross_validate(
   ...     lr,
   ...     X,
   ...     y,
   ...     cv=GroupKFold(),
-  ...     props={"sample_weight": my_weights, "groups": my_groups},
+  ...     params={"sample_weight": my_weights, "groups": my_groups},
   ...     scoring=weighted_acc,
   ... )
 
-If :meth:`linear_model.LogisticRegressionCV.set_fit_request` has not
-been called, :func:`~model_selection.cross_validate` will raise an
-error because ``sample_weight`` is passed in but
-:class:`~linear_model.LogisticRegressionCV` would not be explicitly configured
-to recognize the weights.
+If :meth:`linear_model.LogisticRegressionCV.set_fit_request` had not been called,
+:func:`~model_selection.cross_validate` would raise an error because ``sample_weight``
+is passed but :class:`~linear_model.LogisticRegressionCV` would not be explicitly
+configured to recognize the weights.
 
 Unweighted feature selection
 ----------------------------
 
-Setting request values for metadata are only required if the object, e.g. estimator,
-scorer, etc., is a consumer of that metadata Unlike
+Setting request values for metadata is only required if the object, e.g. estimator,
+scorer, etc., is a potential :term:`consumers <consumer>` of that metadata. Unlike
```

Comment:
not just "potential", they need to set it if they are its consumer and the metadata is passed. Otherwise an error will be raised.